src/main/java/ch/njol/skript/expressions/ExprPermissions.java

```diff
@@ -18,71 +18,146 @@
-@Since("2.2-dev33")
-public class ExprPermissions extends SimpleExpression<String> {
-	
+@Since("2.2-dev33, INSERT VERSION (changers)")
+public class ExprPermissions extends PropertyExpression<Entity, String> {
+
 	static {
-		Skript.registerExpression(ExprPermissions.class, String.class, ExpressionType.PROPERTY, "[(all [[of] the]|the)] permissions (from|of) %players%", "[(all [[of] the]|the)] %players%'[s] permissions");
+		Skript.registerExpression(ExprPermissions.class, String.class, ExpressionType.PROPERTY,
+				"[all [[of] the]|the] permissions (from|of) %entities%",
+				"%entities%'[s] permissions"
+		);
 	}
-	
-	@SuppressWarnings("null")
-	private Expression<Player> players;
-	
-	@SuppressWarnings({"null", "unchecked"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		players = (Expression<Player>) exprs[0];
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr((Expression<Entity>) exprs[0]);
 		return true;
 	}
 
 	@Override
-	@Nullable
-	protected String[] get(Event e) {
-		final Set<String> permissions = new HashSet<>();
-		for (Player player : players.getArray(e))
-			for (final PermissionAttachmentInfo permission : player.getEffectivePermissions())
-				permissions.add(permission.getPermission());
-		return permissions.toArray(new String[permissions.size()]);
+	protected String[] get(Event event, Entity[] source) {
+		return getExpr().stream(event)
+				.flatMap(permissible -> permissible.getEffectivePermissions().stream())
+				.map(PermissionAttachmentInfo::getPermission)
+				.toArray(String[]::new);
 	}
-	
+
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case ADD:
+			case REMOVE:
+				return CollectionUtils.array(String[].class);
+			case DELETE:
+				return CollectionUtils.array();
+			default:
+				return null;
+		}
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		List<String> permissions = new ArrayList<>();
+		if (mode != ChangeMode.DELETE) {
+			assert delta != null;
+			for (Object permission : delta) {
+				if (permission == null)
+					continue;
+				permissions.add((String) permission);
+			}
+		}
+
+		switch (mode) {
+			case DELETE:
+			case REMOVE:
+				for (Entity entity : getExpr().getArray(event)) {
+					for (PermissionAttachmentInfo info : entity.getEffectivePermissions()) {
+						PermissionAttachment attachment = info.getAttachment();
+						if (attachment == null)
+							continue;
+						for (String permission : attachment.getPermissions().keySet()) {
+							if (mode == ChangeMode.DELETE) {
+								attachment.unsetPermission(permission);
+							} else if (permissions.contains(permission)) {
+								attachment.unsetPermission(permission);
+							}
```

Comment:
Is there a reason to check that the permission is present? Can't we just do something like:
```
loop over attachments:
  loop over permissions to remove:
    remove permission from attachment
```
There doesn't seem to be any issue with removing a permission that is not present. It seems like we could make this more efficient